remhos.cpp

```diff
@@ -1333,6 +1465,41 @@ void AdvectionOperator::Mult(const Vector &X, Vector &Y) const
    }
 }
 
+double AdvectionOperator::ComputeTimeStepEstimate(const Vector &x,
+                                                  const Vector &dx,
+                                                  const Vector &x_min,
+                                                  const Vector &x_max) const
+{
+   if (dt_control == TimeStepControl::FixedTimeStep) { return dt; }
+
+   // x_min <= x + dt * dx <= x_max.
+   int n = x.Size();
+   const double cfl = 0.5, eps = 1e-12, dt_min = 1e-10;
+   double dt = numeric_limits<double>::infinity();
+
+   for (int i = 0; i < n; i++)
+   {
+      if (dx(i) > eps)
+      {
+         dt = fmin(dt, (x_max(i) - x(i)) / dx(i) );
+      }
+      else if (dx(i) < -eps)
+      {
+         dt = fmin(dt, (x_min(i) - x(i)) / dx(i) );
+      }
+
+      if (dt < dt_min)
+      {
+         std::cout << dt << " " << dt_min << std::endl;
+         std::cout << "min / max: " << x_min(i) << " " << x_max(i) << std::endl;
+         std::cout << x(i) << " " << dx(i) << " " << x(i) + dt * dx(i) << std::endl;
+         MFEM_ABORT("Time step too small.");
+      }
+   }
+
+   return cfl * dt;
```

Comment:
I think the cfl should be moved inside the loop, otherwise we will keep shrinking the dt every time we call `ComputeTimeStepEstimate` right?